dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/common/DefaultIdentifiableObjectManager.java

```diff
@@ -1138,57 +1165,51 @@ public <T extends IdentifiableObject> List<T> getByAttributeAndValue( Class<T> t
         return store != null && store.isAttributeValueUnique( object, attributeValue );
     }
 
     @Override
     @Transactional( readOnly = true )
-    public <T extends IdentifiableObject> boolean isAttributeValueUnique( Class<T> type,
-        T object, Attribute attribute, String value )
+    public <T extends IdentifiableObject> boolean isAttributeValueUnique( @Nonnull Class<T> type,
+        @Nonnull T object, @Nonnull Attribute attribute, @Nonnull String value )
     {
         IdentifiableObjectStore<T> store = getIdentifiableObjectStore( type );
         return store != null && store.isAttributeValueUnique( object, attribute, value );
     }
 
+    @Nonnull
     @Override
     @Transactional( readOnly = true )
-    public <T extends IdentifiableObject> List<T> getAllByAttributeAndValues( Class<T> type,
-        Attribute attribute, List<String> values )
+    public <T extends IdentifiableObject> List<T> getAllByAttributeAndValues( @Nonnull Class<T> type,
+        @Nonnull Attribute attribute, @Nonnull List<String> values )
     {
         IdentifiableObjectStore<T> store = getIdentifiableObjectStore( type );
         return store != null ? store.getAllByAttributeAndValues( attribute, values ) : List.of();
     }
 
+    @Nonnull
     @Override
     @Transactional( readOnly = true )
     public Map<Class<? extends IdentifiableObject>, IdentifiableObject> getDefaults()
     {
-        Long catId = defaultObjectCache.get( Category.class.getName(),
-            key -> getByName( Category.class, DEFAULT ).getId() );
-        Long cateComboId = defaultObjectCache.get( CategoryCombo.class.getName(),
-            key -> getByName( CategoryCombo.class, DEFAULT ).getId() );
-        Long catOptionId = defaultObjectCache.get( CategoryOption.class.getName(),
-            key -> getByName( CategoryOption.class, DEFAULT ).getId() );
-        Long catOptionComboId = defaultObjectCache.get(
-            CategoryOptionCombo.class.getName(),
-            key -> getByName( CategoryOptionCombo.class, DEFAULT ).getId() );
-
-        return new ImmutableMap.Builder<Class<? extends IdentifiableObject>, IdentifiableObject>()
-            .put( Category.class, Objects.requireNonNull( get( Category.class, catId ) ) )
-            .put( CategoryCombo.class, Objects.requireNonNull( get( CategoryCombo.class, cateComboId ) ) )
-            .put( CategoryOption.class, Objects.requireNonNull( get( CategoryOption.class, catOptionId ) ) )
-            .put( CategoryOptionCombo.class,
-                Objects.requireNonNull( get( CategoryOptionCombo.class, catOptionComboId ) ) )
-            .build();
+        BiFunction<Map<Class<?>, Long>, Class<? extends IdentifiableObject>, Long> getIdByName = ( cache,
+            type ) -> defaultObjectCache.computeIfAbsent( type, key -> {
+                IdentifiableObject obj = getByName( type, DEFAULT );
+                return obj == null ? -1 : obj.getId();
```

Comment:
FYI: For the case where the default object actually does not exist `-1` is returned as ID which later will cause an exception when the ID is used to lookup the object. Effectively a NPE changed to a object not found exception.